CDCgov/prime-reportstream
# Fixed bug 6454 - Duplicates in destinations

See #6454 for details of the bug.
Immediate fix is to add constraints to the joins with the settings table to eliminate deleted and inactive settings.

Test Steps:

Starting with ReportStream running the master branch (NOT this branch), you can reproduce the bug by doing the following:

- Pull the nv-phd.elr receiver setting into a file:
./prime receiver get --name nv-phd.elr > MY-nv-phd.elr.yml

- grab that yml, MAKE A MINOR CHANGE TO THT FILE, and save it.  This will cause the bug, because now there is an old/deleted/inactive setting in the setting table.
./prime receiver set --input MY-nv-phd.elr.yml

- Generate some fake data
./prime data --input-fake 10 --input-schema primedatainput/pdi-covid-19 --target-states NV --output ./junk/nevada.csv

- Submit the fake data
curl -X POST -H "client:simple_report" -H "Content-Type: text/csv" --data-binary "@./junk/nevada.csv" "http://localhost:7071/api/reports"

- YOU WILL SEE THE BUG:  NEVADA is DUPLICATED in the destinations.

- _Now change to this branch and rebuild/rerun RS._

- Now run the above curl again- you should now see good/correct results - only a _single_ destination:   nv-phd.elr.

## Changes
- Corrected the constraints on the joins with Setting table.

## Checklist

### Testing
- [N/A] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [NO] Added tests?

### Process
- [N] Are there licensing issues with any new dependencies introduced?
- [Y] Includes a summary of what a code reviewer should test/verify?
- [N/A] Updated the release notes?
- [N/A] Database changes are submitted as a separate PR?
- [N/A] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #6454 

## To Be Done
- The query as currently constructed is still wrong.    The requirement was to do the history/lineage in two steps:  1) Pull ALL the data, then 2) Walk the data and generate the subset of data that is interesting to the user of this particular API call.   The query as written breaks requirement (1) because it prematurely constrains the result.  The constraint should have been done in step (2).  An example of premature constraint is:   if a receiver has been completely removed from RS, the query as now written will not return their data.
- We need to improve our history API smoke test to actually count the number of destinations to ensure they are correct.  Other tests of overall correctness of the json response should be added as well.  None of that has been done.

## Specific Security-related subjects a reviewer should pay specific attention to
- None.
-
